lib/linguist/heuristics.rb

```diff
@@ -274,6 +274,16 @@ def call(data)
       end
     end
 
+    disambiguate "XML", "Modula-2", "Linux Kernel Module", "AMPL" do |data|
+      if data.include?('<!ENTITY ')
+        Language["XML"]
+      elsif /MODULE\s\w+\s*;/i.match(data) || /^\s*END \w+;$/i.match(data)
+        Language["Modula-2"]
+      else
+        [Language["Linux Kernel Module"], Language["AMPL"]]
```

Comment:
I didn't use a heuristic here because I thought we said that we should only use them as a last resort. The classifier is perfectly capable of distinguishing Linux Kernel Module files from AMPL files.

Of course, @vitaut's suggestion would work. @arfon Do you think I should add it?